### PR TITLE
Implement Split Multiple Declarations Refactoring

### LINKED ISSUE #130: Split into multiple declarations
## Is this request related to a problem? Please describe.

It can happen to have many declarations made together, and we may want to split them so we can treat them individually.

## Describe the solution you'd like

```js
const firstName = "Jane", lastName = "Doe";
```

Should become

```js
const firstName = "Jane";
const lastName = "Doe";
```

Same for `var` and `let` keywords.

Beware of preserving types for TS.

## Additional context

This seems to be an easy refactoring to do: convert an existing Variable Declarator into multiple ones, one per declaration.

### PR DESCRIPTION
**Pull request by iulspop**
Closes #130 

Implemented the split multiple declarations refactoring.
Type annotations are preserved correctly for Typescript.

split-multiple-declarations

I'm not sure if the additions to the `package.json` are correct

### CODE REVIEW

**Comment by iulspop** [comment]
Is there anything else I should add to the pull request description?

**Comment by nicoespeon** [comment]
@iulspop it's perfect! Thank you for the PR, I'll review it this week 👍

File: README.md [hunk 1]
```diff
@@ -484,6 +485,20 @@ Splits the declaration of the variable and its initialization. If it's a `const`
 
 [⬆️ Go to Table of Contents](#table-of-contents)
 
+### Split Multiple Declarations
+
+> 💡 Available as Quick Fix (`Alt ↵`)
+
+Splits multiple variables declarated together onto a single line each.
```
Comment (nicoespeon) [inline README.md:492]: 👍 Very great, thanks for adding docs too

File: package.json [hunk 2]
```diff
@@ -1177,6 +1183,26 @@
...
+          "command": "abracadabra.splitMultipleDeclarations",
+          "when": "editorLangId == typescript"
+        },
+        {
+          "command": "abracadabra.splitMultipleDeclarations",
+          "when": "editorLangId == typescriptreact"
+        },
+        {
+          "command": "abracadabra.splitMultipleDeclarations",
+          "when": "editorLangId == vue"
```
Comment (nicoespeon) [inline package.json:1205]: 👍 Looks good. I agree that this part is a bit tricky and requires a lot of boilerplate. We'll eventually automate this so it's less error-prone.

File: src/extension.ts [hunk 3]
```diff
@@ -35,9 +35,10 @@ import removeDeadCode from "./refactorings/remove-dead-code";
 import removeRedundantElse from "./refactorings/remove-redundant-else";
 import renameSymbol from "./refactorings/rename-symbol";
 import replaceBinaryWithAssignment from "./refactorings/replace-binary-with-assignment";
+import simplifyTernary from "./refactorings/simplify-ternary";
```
Comment (nicoespeon) [inline src/extension.ts:38]: 👍 Thanks for re-ordering this one

File: src/refactorings/split-multiple-declarations/split-multiple-declarations.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        expected: `let firstName = 'John';
+let lastName;
+let details = {age: 10, country: "Moon"};`
+      },
+      {
+        description:
+          "typescript multiple declarations split conserves type annotations",
+        code: `let firstName: string = 'John', age: number = 7`,
+        expected: `let firstName: string = 'John';
+let age: number = 7;`
```
Comment (nicoespeon) [inline src/refactorings/split-multiple-declarations/split-multiple-declarations.test.ts:42]: 👍 Perfect, you covered a lot of cases here!

**Review by nicoespeon — APPROVED** [review]
Everything looks great. Thanks for adding the docs and all these tests 👍

**Comment by nicoespeon** [comment]
@allcontributors please add iulspop for code, doc

**Pull request merged**